### PR TITLE
Feature/pvp system

### PR DESCRIPTION
**Pull request by Raycoms**
Adds the first part of complete PVP system.
This contains:

# Changes proposed in this pull request:
- A configuration value to turn this on and off
- A way to identify teams of guards and select their color
- Permission handling to allow griefing under certain conditions
- TownHall handling to delete the colony if destroyed.

Review please

### CODE REVIEW

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java [hunk 1]
```diff
@@ -66,7 +67,7 @@ private void initBlock()
         //Blast resistance for creepers etc. makes them explosion proof
         setResistance(RESISTANCE);
         //Hardness of 10 takes a long time to mine to not loose progress
-        setHardness(HARDNESS);
+        setHardness(Configurations.gameplay.pvp_mode ? HARDNESS * 2 : HARDNESS);
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java]: Bumb this factor to 4. It should be a challenge.

File: src/main/java/com/minecolonies/coremod/items/ItemChiefSword.java [hunk 2]
```diff
@@ -9,26 +9,19 @@
...
 import org.jetbrains.annotations.NotNull;
 
 import java.util.stream.Stream;
 
+import static com.minecolonies.api.util.constant.Constants.*;
+
 /**
  * Class handling the Chief Sword item.
  */
 public class ItemChiefSword extends ItemSword
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/items/ItemChiefSword.java:23]: Maybe buff the ChiefSword in PVP Mode?

**Review by marchermans — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/items/ItemChiefSword.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/items/ItemChiefSword.java:23]: Nah, it's quite rare and it's less op than most tinkers swords as well, so I don't see much of an issue with it.

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java]: good idea

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Pull request merged**